### PR TITLE
Actions/GitHub actions

### PR DESCRIPTION
**Pull request by jared-w**
This supersedes and closes #775

closes #775

### CODE REVIEW

File: .github/PropertiesTest.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+public class PropertiesTest {
+    public static void main(String[] args)
+        throws Exception {
+        String value = System.getProperty("sun.boot.class.path");
+        System.out.println(value);
+    }
+}
```
Comment (atomb) [inline .github/PropertiesTest.java:7]: It had honestly never occurred to me to just do this to find the classpath. Brilliant!

Comment (jared-w) [reply .github/PropertiesTest.java:7]: Found this randomly buried inside of a stackoverflow answer, actually. Standing on the shoulders of giants and all that :)

**Comment by jared-w** [comment]
We can start abstracting out common actions eventually. A hypothetical `GaloisInc/actions/install-solvers` could install solvers more efficiently and internally setup caches to not have to re-download things, `GaloisInc/actions/release` could abstract out some of the release logic. `GaloisInc/actions/haskell-setup` could abstract out our `cabal.GHC-x.x.x.config` -> `cabal.project.freeze` conventions... And so on.

Of course, I'd hate to pre-optimize before I implement a few more CI pipelines for various projects. But so far it's looking quite promising

**Comment by atomb** [comment]
Excellent. That's what I was hoping. And, yes, definitely worth sampling a few more instances before generalizing.

**Pull request merged**